crates/fuel-core/src/graphql_api.rs

```diff
@@ -13,13 +16,58 @@ pub(crate) mod view_extension;
 pub mod worker_service;
 
 #[derive(Clone, Debug)]
-pub struct Config {
+pub struct ServiceConfig {
     pub addr: SocketAddr,
+    pub max_queries_depth: usize,
+    pub max_queries_complexity: usize,
+    pub max_queries_recursive_depth: usize,
+    /// Time to wait after submitting a query before debug info will be logged about query.
+    pub query_log_threshold_time: Duration,
+    pub api_request_timeout: Duration,
+}
+
+pub struct Costs {
+    pub balance_query: usize,
+    pub coins_to_spend: usize,
+    pub get_peers: usize,
+    pub estimate_predicates: usize,
+    pub dry_run: usize,
+    pub submit: usize,
+    pub submit_and_await: usize,
+    pub status_change: usize,
+    pub raw_payload: usize,
+    pub storage_read: usize,
+    pub storage_iterator: usize,
+    pub bytecode_read: usize,
+}
+
+pub const QUERY_COSTS: Costs = Costs {
+    // balance_query: 4000,
+    balance_query: 10001,
```

Comment:
[nit] maybe we should have a constant for the default max complexity, and then use that here (ie. MAX_COMPLEXITY_DEFAULT / 2 + 1)